src/nrtk_explorer/library/object_detector.py

```diff
@@ -77,17 +78,38 @@ def eval(
             batches[img.size][0].append(path)
             batches[img.size][1].append(img)
 
-        predictions_in_baches = [
-            zip(
-                image_ids,
-                self.pipeline(images, batch_size=batch_size),
-            )
-            for image_ids, images in batches.values()
-        ]
-
-        predictions_by_image_id = {
-            image_id: predictions
-            for batch in predictions_in_baches
-            for image_id, predictions in batch
-        }
-        return predictions_by_image_id
+        adjusted_batch_size = batch_size
+        while adjusted_batch_size >= 2:
+            try:
+                predictions_in_baches = [
+                    zip(
+                        image_ids,
+                        self.pipeline(images, batch_size=adjusted_batch_size),
+                    )
+                    for image_ids, images in batches.values()
+                ]
+
+                predictions_by_image_id = {
+                    image_id: predictions
+                    for batch in predictions_in_baches
+                    for image_id, predictions in batch
+                }
+                return predictions_by_image_id
+
+            except RuntimeError as e:
+                if "out of memory" in str(e) and adjusted_batch_size > 2:
+                    previous_batch_size = adjusted_batch_size
+                    adjusted_batch_size = max(2, adjusted_batch_size // 2)
```

Comment:
```suggestion
                    adjusted_batch_size =  adjusted_batch_size // 2
```
We want to allow batch size of 1.